### ISSUE TITLE
Cheerio failing with lax HTML structure?

### ISSUE THREAD

**Issue by ajmas** [cheeriojs/cheerio#826]
I have a page which I am trying to scape of the form:

```
<tr><td><a href="...">mylink</a></td></tr>
<tr><td><a href="...">mylink</a></td></td></tr>
<tr><td><a href="...">mylink</a></td></td></tr>
```

Which seems to cause issues with the following logic:

```
var nodes = $('div table[cellpadding=3] tr td a')
```

The result here is one match, as opposed to three. I didn't see anything in the cheerio docs on how to normalise the HTML, so can anyone indicate whether this is something where depend on another library to normalise or whether cheerio should deal with it. Note, if I take out the extra </td> then it returns the right number of anchor tags.

Note, that JQuery handles this in Google Chrome, but it would appear this is probably because browsers normalise the HTML prior to rendering?

**Comment by luanmuniz** [comment]
Actually Chrome is the one that "normalize" the HTML. jQuery just read the DOM good to go. It doesn't parse the HTML like `cheerio`.

The module used to parse the HTML is `htmlparser2` and cheerio just read the result from it and apply a nice api. Maybe its better to post the issue there.

I don't know if this solve your problem, but maybe doing something like `var $ = cherrio.load(html.replace('</td></td>', '</td>'))` or maybe doing something more complex with a regex looking for 2 equals closing tags together...

**Comment by ajmas** [comment]
I ended up using jsdom, before assigning the data to cheerio. I am still using cheerio, since I still like it, even if this may see what heavy. 

```
function processPage(data) {
    data = jsdom(data).documentElement.outerHTML
    $ = cheerio.load(data);
    ...
}
```

**Comment by fb55** [comment]
Fixed by #985
